websites/Z/Zoro/presence.ts

```diff
@@ -0,0 +1,174 @@
+      break;
+    }
+    case pathname.startsWith("/user") && pathname: {
+      const profile: HTMLDivElement = document.querySelector("div.ph-title"),
+        link: HTMLAnchorElement = document
+          .querySelector("ul.nav.nav-tabs.pre-tabs")
+          .querySelector("a.nav-link.active");
+      if (profile) presenceData.details = `Viewing User: ${profile.innerText}`;
+      if (link) presenceData.state = `At ${link.innerText}`;
+      break;
+    }
+    case (pathname.startsWith("people") || pathname.startsWith("character")) &&
+      pathname: {
+      const name: HTMLHeadingElement = document.querySelector("h4.name");
+      if (name) {
+        presenceData.details = `Looking at ${
+          pathname.startsWith("/people") ? "People" : "Character"
+        }`;
+        presenceData.state = name.innerText;
+      }
+      break;
+    }
+    case pathname.startsWith("/az-list") && pathname: {
+      presenceData.details = "Looking at Anime list";
+      if (pathname !== "/az-list") {
+        presenceData.state = `Titles starting with ${
+          pathname.substring(9) === "other"
+            ? "Other characters"
+            : `${pathname.substring(9)}`
+        }`;
+      }
+      break;
+    }
+    case pathname.startsWith("/watch2gether/") && pathname: {
+      if (pathname === "/watch2gether/")
+        presenceData.details = "Looking for anime rooms";
+      else {
+        const filmName: HTMLHeadingElement =
+          document.querySelector("h2.film-name");
+        presenceData.details = "In a room";
+        if (filmName) presenceData.state = `Watching ${filmName.innerText}`;
+        if (data) {
+          if (data.iFrameVideo && !data.paused) {
+            [, presenceData.endTimestamp] = presence.getTimestamps(
+              data.currTime,
+              data.duration
+            );
+          }
+        }
+        presenceData.buttons = [
+          {
+            label: "Join Room",
+            url: document.location.href
+          }
+        ];
+      }
+      break;
+    }
+    case pathname.startsWith("/watch") && pathname: {
+      const title: HTMLDataListElement = document.querySelector(
+        "li.breadcrumb-item.dynamic-name.active"
+      ),
+        episode: HTMLSpanElement = document.querySelector(
+          "span#cm-episode-number"
+        );
+      if (title) presenceData.details = title.innerText;
+      if (episode) presenceData.state = `Episode ${episode.innerText}`;
+      if (data) {
+        if (data.iFrameVideo && !data.paused) {
+          [, presenceData.endTimestamp] = presence.getTimestamps(
+            data.currTime,
+            data.duration
+          );
+        }
+      }
+      presenceData.buttons = [
+        {
+          label: "Watch Episode",
+          url: document.location.href
+        }
+      ];
+      break;
+    }
+    case "/events": {
+      presenceData.details = "Looking at events";
+      break;
+    }
+    case pathname.startsWith("/event/") && pathname: {
```

Comment:
this case statement doesn't have any sense. You are checking if pathname starts with "/event" so basically it is already a thing?
Do
```suggestion
    case pathname?.startsWith("/event/"): {
```

Reply:
>You are checking if pathname starts with "/event" so basically it is already a thing

Yes, but we need to return a string, as we are matching against `pathname` a string, `String.startsWith` returns a `boolean`.

[Check this](https://stackoverflow.com/a/43398039)